client/src/templates/Challenges/components/interactive-editor.tsx

```diff
@@ -43,7 +43,12 @@ const InteractiveEditor = ({ files }: Props) => {
     return files.some(f => f.ext === ext);
   }
 
-  const showConsole = got('js') || got('ts');
+  const hasHtmlOrCss = got('html') || got('css');
+  const hasScript = got('js') || got('ts');
+  const isPureScript = hasScript && !hasHtmlOrCss;
+
+  const layout = isPureScript ? 'console' : 'preview';
+  const showConsole = layout === 'preview' && hasScript;
```

Comment:
```suggestion
const layout = hasHtmlOrCss ? 'preview' : 'console';
const showConsole = hasHtmlOrCss && hasScript;
```

Keep it simple, and readable.